src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+    const last = this[this.length - 1];
+
+    for (const item of this) {
+      if (item === null || item === undefined) {
+        string += '';
+      } else {
+        string += item;
+      }
```

Comment:
just use
```suggestion
      if (item !== null && item !== undefined) {
        string += item;
      }
```

Reply:
I tried once more => 4 tests are failed. 
![image](https://user-images.githubusercontent.com/119630749/213763293-eece049b-8a82-4540-b987-e363830e8a65.png)

Reply:
Look carefully - not `||` but `&&`.
And all tests will pass.

Reply:
Thank you! :)  Done